components/kyma-environment-broker/internal/broker/instance_update.go

```diff
@@ -2,26 +2,60 @@ package broker
 
 import (
 	"context"
+	"encoding/json"
 	"errors"
 
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
 	"github.com/pivotal-cf/brokerapi/v7/domain"
 	"github.com/sirupsen/logrus"
 )
 
 type UpdateEndpoint struct {
 	log logrus.FieldLogger
+
+	instanceStorage storage.Instances
 }
 
-func NewUpdate(log logrus.FieldLogger) *UpdateEndpoint {
-	return &UpdateEndpoint{log: log.WithField("service", "UpdateEndpoint")}
+func NewUpdate(instanceStorage storage.Instances, log logrus.FieldLogger) *UpdateEndpoint {
+	return &UpdateEndpoint{
+		log:             log.WithField("service", "UpdateEndpoint"),
+		instanceStorage: instanceStorage,
+	}
 }
 
 // Update modifies an existing service instance
 //  PATCH /v2/service_instances/{instance_id}
 func (b *UpdateEndpoint) Update(ctx context.Context, instanceID string, details domain.UpdateDetails, asyncAllowed bool) (domain.UpdateServiceSpec, error) {
-	b.log.Infof("Update instanceID: %s", instanceID)
-	b.log.Infof("Update details: %+v", details)
-	b.log.Infof("Update asyncAllowed: %v", asyncAllowed)
+	logger := b.log.WithField("instanceID", instanceID)
+	logger.Infof("Update instanceID: %s", instanceID)
+	logger.Infof("Update asyncAllowed: %v", asyncAllowed)
+
+	instance, err := b.instanceStorage.GetByID(instanceID)
+	if err != nil {
+		logger.Errorf("unable to get instance")
```

Comment:
What do you think about logging the error that we get from the `GetByID` method too?